modal/mount.py

```diff
@@ -810,3 +810,109 @@ def _is_modal_path(remote_path: PurePosixPath):
         if is_modal_path:
             return True
     return False
+
+REMOTE_PACKAGES_PATH = "/__modal/deps"
+REMOTE_SITECUSTOMIZE_PATH = "/pkg/sitecustomize.py"
+
+SITECUSTOMIZE_CONTENT = f"""
+# This file is automatically generated by Modal.
+# It ensures that Modal's python dependencies are available in the Python PATH,
+# while prioritizing user-installed packages.
+import sys; sys.path.append('{REMOTE_PACKAGES_PATH}')
+""".strip()
+
+async def _create_single_mount(
+    client: _Client,
+    builder_version: str,
+    python_version: str,
+    platform: str,
+    arch: str,
+    uv_python_platform: str = None,
+    check_if_exists: bool = True,
+):
+    import subprocess
+    import tempfile
+
+    profile_environment = config.get("environment")
+    abi_tag = "cp" + python_version.replace(".", "")
+    mount_name = f"{builder_version}-{abi_tag}-{platform}-{arch}"
+    uv_python_platform = uv_python_platform or f"{arch}-{platform}"
+
+    if check_if_exists:
+        try:
+            Mount.from_name(mount_name, namespace=api_pb2.DEPLOYMENT_NAMESPACE_GLOBAL).hydrate(client)
```

Comment:
If this is an async function I think you'll want to be using the `.aio` interface on `.hydrate`.